durable-storage/src/repo.rs

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+
+use tempfile::TempDir;
+
+#[derive(Debug, thiserror::Error)]
+pub enum DirectoryManagerError {
+    #[error("Failed instantiating a new directory manager: {0}")]
+    New(io::Error),
+
+    #[error("Failed creating a new checkpoint directory: {0}")]
+    Checkpoint(io::Error),
+}
+
+/// The [`DirectoryManager`] represents the root directory where commitments & internal data should
+/// be stored.
+pub struct DirectoryManager {
+    path: PathBuf,
+}
+
+impl DirectoryManager {
+    /// Instantiate a new [`DirectoryManager`] at the given `path`.
+    ///
+    /// - If the folder is not existent, it will be created and the [`DirectoryManager`] will be
+    ///   empty.
+    /// - If the folder exists, it will assume a valid repository for the paths managed, i.e.
+    ///   `<repo>/temporary/db_<id>/checkpoint` is a valid rocksdb checkpoint.
+    pub fn new(path: &Path) -> Result<Self, DirectoryManagerError> {
+        let temp_dir = path.join("temporary");
```

Comment:
Could use the `tempdir` crate here too.

Reply:
Also, multiple `DirectoryManager`'s can be instantiated in the same directory, so deleting the `temporary` subfolder on `drop` won't be correct in this case